libraries/standard/mqtt/include/mqtt_lightweight.h

```diff
@@ -22,18 +22,18 @@
 #ifndef MQTT_LIGHTWEIGHT_H
 #define MQTT_LIGHTWEIGHT_H
 
+#include <stddef.h>
+#include <stdint.h>
+
 /* bools are only defined in C99+ */
-#if defined( __cplusplus ) || __STDC_VERSION__ >= 199901L
+#if defined( __cplusplus ) || ( defined( __STDC_VERSION__ ) && ( __STDC_VERSION__ >= 199901L ) )
     #include <stdbool.h>
 #elif !defined( bool )
-    #define bool     signed char
-    #define false    0
-    #define true     1
+    #define bool     int8_t
+    #define false    ( int8_t ) 0
+    #define true     ( int8_t ) 1
```

Comment:
I think some wanted to get rid of our `#define` of `bool`, and others wanted to keep it. I don't believe there was a clear decision on that matter, and I don't think removing it is in scope for this PR. Note that due to the changes in this PR, removal (if desired) should be straightforward, just a search and replace of `bool` with the desired type, `false` with `0`, and `true` with `1`